src/core/worker.lua

```diff
@@ -0,0 +1,145 @@
+-- worker.lua - Execute "worker" child processes to execute app networks
+-- Use of this source code is governed by the Apache 2.0 license; see COPYING.
+
+module(..., package.seeall)
+
+-- API:
+-- start(name, core)
+-- stop(name)
+-- status() -> table of { name = <info> }
+-- configure(name, config)
+
+local lib = require("core.lib")
+local shm = require("core.shm")
+local S = require("syscall")
+
+--------------------------------------------------------------
+-- Master (parent) process code
+--------------------------------------------------------------
+
+local children = {}
+
+local function child (name)
+   return children[name] or error("no such child: " .. name)
+end
+
+-- Start a worker process with affinity to a specific CPU core.
+-- The child will execute an app network when provided with configure().
+function start (name, core)
+   local pid = S.fork()
+   if pid == 0 then
+      -- Lock affinity for this child
+      S.sched_setaffinity(0, {core})
+      local env = { "SNABB_PROGRAM_LUACODE=require('core.worker').init()",
+                    "SNABB_WORKER_NAME="..name,
+                    "SNABB_WORKER_PARENT="..S.getppid() }
+      -- /proc/$$/exe is a link to the same Snabb executable that we are running
+      S.execve(("/proc/%d/exe"):format(S.getpid()), {}, env)
```

Comment:
This gives me the shivers -- what if someone upgraded this snabb in place?  What if it was moved or made clean?  In any case we need an assert and anerror message so we don't fall off and keep running the parent.